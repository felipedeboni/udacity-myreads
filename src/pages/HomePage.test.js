import React from 'react';
import HomePage from './HomePage';
import { BOOKSHELVES } from '../utils/constants';
import { shallow } from 'enzyme';

const books = [];
it('renders without crashing', () => {
  const test = <HomePage books={books} onMoveToShelf={jest.fn()} />;

  expect(shallow(test)).toMatchSnapshot();
});

it('should have a link to the search page', () => {
  const test = shallow(<HomePage books={books} onMoveToShelf={jest.fn()} />);

  expect(test.find('Link[to="/search"]').exists()).toBe(true);
});

it('should show bookshelves', () => {
  const test = shallow(<HomePage books={books} onMoveToShelf={jest.fn()} />);

  for (let bookshelf of BOOKSHELVES) {
    expect(test.find(`Bookshelf[id="${bookshelf.id}"]`).exists()).toBe(true);
  }
});
