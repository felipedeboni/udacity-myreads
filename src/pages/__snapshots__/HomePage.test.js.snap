// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="list-books"
>
  <div
    className="list-books-title"
  >
    <h1>
      MyReads
    </h1>
  </div>
  <div
    className="list-books-content"
  >
    <Bookshelf
      books={Array []}
      id="currentlyReading"
      key="0"
      onMoveToShelf={[Function]}
      title="Currently Reading"
    />
    <Bookshelf
      books={Array []}
      id="wantToRead"
      key="1"
      onMoveToShelf={[Function]}
      title="Want to Read"
    />
    <Bookshelf
      books={Array []}
      id="read"
      key="2"
      onMoveToShelf={[Function]}
      title="Read"
    />
  </div>
  <div
    className="open-search"
  >
    <Link
      replace={false}
      to="/search"
    >
      Add a book
    </Link>
  </div>
</div>
`;
